components/script/dom/htmltablesectionelement.rs

```diff
@@ -71,6 +76,64 @@ impl HTMLTableSectionElementMethods for HTMLTableSectionElement {
     fn Rows(&self) -> Root<HTMLCollection> {
         HTMLCollection::create(&window_from_node(self), NodeCast::from_ref(self), box RowsFilter)
     }
+
+    // https://html.spec.whatwg.org/multipage/#dom-tbody-insertrow
+    fn InsertRow(&self, index: i32) -> Fallible<Root<HTMLElement>> {
+        if index < -1 {
+            return Err(Error::IndexSize);
+        }
+
+        let self_node = NodeCast::from_ref(self);
+        let tr = HTMLTableRowElement::new(String::from("tr"), None, self_node.owner_doc().r());
+
+        if index == -1 {
+            try!(self_node.AppendChild(NodeCast::from_ref(tr.r())));
+            return Ok(HTMLElementCast::from_root(tr));
+        }
+
+        let mut i = 0;
+        for element in self.Rows().elements_iter() {
+            if i == index {
+                try!(self_node.InsertBefore(NodeCast::from_ref(tr.r()),
+                                            Some(NodeCast::from_root(element)).r()));
+                return Ok(HTMLElementCast::from_root(tr));
+            }
+            i += 1;
+        }
+
+        if i == index {
+            try!(self_node.AppendChild(NodeCast::from_ref(tr.r())));
+            return Ok(HTMLElementCast::from_root(tr));
+        }
+
+        return Err(Error::IndexSize)
```

Comment:
So I saw your solution, but decided I wasn't a fan of the nested `Option`s, so I changed it up a little bit. Check out my latest force push and let me know what you think